### PR TITLE
Replace custom filter edit

### PR DESCRIPTION
**Pull request by JSReds**
Replace custom filter Edit with Form

Change-type: minor
Signed-off-by: Andrea Rosci <andrear@balena.io>

work in progress to accept array types RJSForm: https://github.com/rjsf-team/react-jsonschema-form/issues/282 (partially solved)

Screenshot 2022-02-15 at 13 56 03
Screenshot 2022-02-15 at 13 56 16
Screenshot 2022-02-15 at 13 56 36
Screenshot 2022-02-15 at 13 56 43

---
##### Contributor checklist

- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

**Comment by LucianBuzzo** [comment]
This is a great change! It looks like there are a few issues where the `Form` component doesn't work with values that the previous edit types could handle. I think `type: ['string', null]` is an example of this. We can use this PR as a forcing function to make the `Form` component more capable.

File: src/components/Filters/FilterModal.tsx [hunk 1]
```diff
@@ -50,22 +48,27 @@ const defaultFilterCompareFn: FilterFieldCompareFn = (a, b) => {
 };
 
 const FilterInput = (props: FilterInputProps) => {
+	const { schema, value, onUpdate } = props;
 	const model = getDataModel(props.schema);
-
-	if (!model) {
-		return null;
```
Comment (thgreasi) [inline src/components/Filters/FilterModal.tsx:56]: Maybe `if (!model) return null;` is fine to keep?
My concern is whether the filters will be able to work if we find a type for which we don't have a DataModel for, even though the Form will be able to render it.

File: src/components/Filters/FilterModal.tsx [hunk 2]
```diff
@@ -50,22 +48,27 @@ const defaultFilterCompareFn: FilterFieldCompareFn = (a, b) => {
 };
 
 const FilterInput = (props: FilterInputProps) => {
+	const { schema, value, onUpdate } = props;
```
Comment (thgreasi) [inline src/components/Filters/FilterModal.tsx:51]: I guess we can inline the destructuring and stop using `props.X` completely :)
```suggestion
const FilterInput = ({ schema, value, onUpdate }: FilterInputProps) => {
```

**Review by thgreasi — COMMENTED** [review]
This looks like magic :tada:

File: src/components/Filters/FilterModal.tsx (hunk 1 above)
Comment (JSReds) [reply src/components/Filters/FilterModal.tsx:56]: if model is not defined we should try to render the form in any case no ? The model is not needed to render the Form component, right ?

File: src/components/Filters/FilterModal.tsx [hunk 3]
```diff
@@ -197,14 +200,14 @@ export const FilterModal = ({
 									/>
 								</Box>
 							)}
-							<Box flex={1}>
+							<Flex flex={1} className="stocazzo" alignItems="center">
```
[the remark this answers is not in the document]
Comment (JSReds) [reply src/components/Filters/FilterModal.tsx:203]: 🤦

**Review by thgreasi — APPROVED** [review]
That's just magic :sparkles:  :tada: :+1:

**Pull request merged**